app/services/calculation_orchestrator.rb

```diff
@@ -8,12 +8,57 @@ class << self
     def call(cpd_lead_provider:,
              contract:,
              aggregator: ::ParticipantEventAggregator,
-             uplift_aggregator: ::ParticipantUpliftAggregator,
              calculator: ::PaymentCalculator::Ecf::PaymentCalculation,
              event_type: :started)
-      total_participants = aggregator.call({ cpd_lead_provider: cpd_lead_provider }, event_type: event_type)
-      uplift_participants = uplift_aggregator.call({ cpd_lead_provider: cpd_lead_provider }, event_type: event_type)
-      calculator.call(contract: contract, total_participants: total_participants, uplift_participants: uplift_participants, event_type: event_type)
+      new(
+        cpd_lead_provider: cpd_lead_provider,
+        contract: contract,
+        aggregator: aggregator,
+        calculator: calculator,
+      ).call(event_type: event_type)
+    end
+  end
+
+  def call(event_type:)
+    calculator.call(contract: contract, aggregations: aggregations(event_type: event_type), event_type: event_type)
+  end
+
+private
+
+  attr_accessor :cpd_lead_provider, :contract, :aggregator, :calculator
+
+  def initialize(cpd_lead_provider:,
+                 contract:,
+                 aggregator: ::ParticipantEventAggregator,
+                 calculator: ::PaymentCalculator::Ecf::PaymentCalculation)
+    @cpd_lead_provider = cpd_lead_provider
+    @contract = contract
+    @aggregator = aggregator
+    @calculator = calculator
+  end
+
+  def aggregators(event_type:)
+    @aggregators ||= Hash.new { |hash, key| hash[key] = aggregate(aggregation_type: key, event_type: event_type) }
+  end
```

Comment:
Yes. It's creating the hash with an initialiser that knows how to get the values of any keys that it doesn't have. Since aggregations could be an expensive operation, they should only be called when required.